components/sup/src/manager/mod.rs

```diff
@@ -723,6 +768,20 @@ impl Manager {
                 let time_to_wait = next_check - now;
                 thread::sleep(time_to_wait.to_std().unwrap());
             }
+
+            // Measure CPU time every second
+            if SteadyTime::now() >= next_cpu_measurement {
+                let current_thread = thread::current();
+                let thread_name = current_thread.name().unwrap();
+                let cpu_duration = cpu_start.elapsed();
+                let cpu_time: f64 = (cpu_duration.as_secs() as f64)
+                    + (cpu_duration.subsec_nanos() as f64) / (1_000_000_000 as f64);
+                CPU_TIME.with_label_values(&[thread_name]).set(cpu_time);
```

Comment:
Will do

Reply:
I don't think so. The original code was representing seconds with an `f64`, but my suggestion was using a `i64` representing nanoseconds. Converting seconds to nanoseconds should be a multiply, rather than converting nanoseconds to seconds with a divide.

Reply:
Ah you're right, my mistake.